### DISCUSSION TITLE
Questions and concerns about backup and borgbackup

### DISCUSSION THREAD

**Discussion by ShapeShifter499** [nextcloud/all-in-one#3630] [Questions] [answered]
If I chose to store the data for Nextcloud on a external disk during install does the borgbackup include that data?

Does borgbackup include user data by default?

I have a 12tb external hardware RAID box that is already set up to be backed up in entirety to a cloud provider with duplicati. I pointed borgbackup to store the files there `/mnt/raid_mount/ncbackup`. I pointed Nextcloud data to `/mnt/raid_mount/ncdata`. 

I only want configuration and the database backed up. I don't need the user data backed up twice.

**Comment by ShapeShifter499** [comment]
I looked into all of this and I see that the included borgbackup will try to back up user data with everything even if it's pointed at a different location than docker like a external drive. It just won't backup any extra data that you include after initial setup like having nextcloud use external drives/mounts (other than the path you used for user data) for extra storage.

Can we get a option that skips backing up user data?   It would be nice to still get configuration and database backup with the option to turn on automatic updates.

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

**Reply by ShapeShifter499** [reply]
>However note that doing this is disrecommended since you will not be able to easily create and restore a backup from the AIO interface anymore

Well my thinking here is one could have had Nextcloud AIO automatically copy backups of the working config and database out to a chosen location that gets automatically backed up. 

But what you're suggesting is will require me to set up an external script that will: stop the instance of Nextcloud AIO, backup the directory of the config and database, trigger updating containers, and restart the instance.

**Reply by szaimen** [reply]
> Well my thinking here is one could have had Nextcloud AIO automatically copy backups of the working config and database out to a chosen location that gets automatically backed up.

Yes but this is not really going to work with Nextcloud as everything including the datadir needs to be restored at the same time. Otherwise the database and the data will get out of sync. (of course you can work around this but it would make the included backup solution much more unreliable.)

> But what you're suggesting is will require me to set up an external script that will: stop the instance of Nextcloud AIO, backup the directory of the config and database, trigger updating containers, and restart the instance.

Yes indeed.

**Reply by ShapeShifter499** [reply]
I guess it works even if it seems a bit clunky. Thank you for the help.
